SPECS/mozjs/mozjs.spec

```
@@ -0,0 +1,116 @@
+
+%global	major 78
+Summary:       Mozilla's JavaScript engine.
+Name:          mozjs
+Version:       78.10.0
+Release:       2%{?dist}
+Group:         Applications/System
+Vendor:        Microsoft Corporation
+License:       MPLv2.0 and MPLv1.1 and BSD and GPLv2+ and GPLv3+ and LGPLv2+ and AFL and ASL 2.0 and CCO
+URL:           https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
+Source0:       https://ftp.mozilla.org/pub/firefox/releases/%{version}esr/source/firefox-%{version}esr.source.tar.xz
+Patch0:        emitter.patch
+Patch1:        emitter_test.patch
+# Build fixes
+Patch2:        init_patch.patch
+Patch3:        spidermonkey_checks_disable.patch
+Distribution:  Mariner
+BuildRequires: which
+BuildRequires: python3-xml
+BuildRequires: python3-libs
+BuildRequires: python3-devel
+BuildRequires: zlib-devel
+BuildRequires: gcc
+BuildRequires: llvm
+BuildRequires: llvm-devel
+BuildRequires: icu-devel
+BuildRequires: rust
+BuildRequires: autoconf213
+Requires:      icu
+Requires:      python3
+Requires:      python3-libs
+Obsoletes:     mozjs60
+Obsoletes:     mozjs68
+Obsoletes:     js
+Provides:      mozjs%{major}
+
+%description
+Mozilla's JavaScript engine includes a just-in-time compiler (JIT) that compiles
+JavaScript to machine code, for a significant speed increase.
+
+%package       devel
+Summary:       mozjs devel
+Group:         Development/Tools
+Requires:      %{name} = %{version}-%{release}
+
+%description   devel
+This contains development tools and libraries for SpiderMonkey.
+
+%prep
+%autosetup -p1 -n firefox-%{version}
+rm -rf modules/zlib
+
+%build
+cd js/src
+%configure \
+    --with-system-icu \
+    --enable-readline \
+    --disable-jemalloc \
+    --disable-tests \
```

Nit: we could try enabling tests, while we're here.

```suggestion
%if %{with_check}
    --enable-tests \
%else
    --disable-tests \
%endif
```

The `%check` section would probably look something like this:

```bash
%check
cd js/src
%make_build test
```

It might be `%make_build check` instead of `%make_build test`.